android/src/main/java/com/imagepicker/Utils.java

```diff
@@ -335,31 +339,65 @@ public static boolean isCameraPermissionFulfilled(Context context, Activity acti
         }
     }
 
-    static ReadableMap getResponseMap(Uri uri, Options options, Context context) {
-        String fileName = uri.getLastPathSegment();
-        int[] dimensions = getImageDimensions(uri, context);
+    static List<Uri> collectUrisFromClipData(ClipData clipData) {
+        List<Uri> fileUris = new ArrayList<>(clipData.getItemCount());
 
-        WritableMap map = Arguments.createMap();
-        map.putString("uri", uri.toString());
-        map.putDouble("fileSize", getFileSize(uri, context));
-        map.putString("fileName", fileName);
-        map.putString("type", getMimeTypeFromFileUri(uri));
-        map.putInt("width", dimensions[0]);
-        map.putInt("height", dimensions[1]);
-
-        if (options.includeBase64) {
-            map.putString("base64", getBase64String(uri, context));
+        for (int i = 0; i < clipData.getItemCount(); ++i) {
+            fileUris.add(clipData.getItemAt(i).getUri());
+        }
+
+        return fileUris;
+    }
+
+    static ReadableMap getResponseMap(List<Uri> uris, Options options, Context context) {
```

Comment:
My gut says we should leave the original methods as is (i.e. without the list of uris) and create new functions that wrap them (i.e. `getImagesResponseMap` note the plural images).

Reply:
makes sense, also as a preparation for integrating it with mixed assets!